spec/requests/unpublishing_spec.rb

```diff
@@ -107,46 +119,60 @@
+    shared_examples "unpublishing with redirects" do
+      it "creates an Unpublishing" do
+        post "/v2/content/#{content_id}/unpublish", params: redirect_params
 
-      expect(response.status).to eq(200), response.body
+        expect(response.status).to eq(200), response.body
 
-      unpublishing = Unpublishing.find_by(edition: edition)
-      expect(unpublishing.type).to eq("redirect")
-      expect(unpublishing.alternative_path).to eq("/new-path")
-    end
+        unpublishing = Unpublishing.find_by(edition: edition)
+        expect(unpublishing.type).to eq("redirect")
+        expect(unpublishing.redirects).to match_array([
+          a_hash_including(destination: "/new-path")
+        ])
+      end
 
-    it "sends a redirect to the live content store" do
-      Timecop.freeze do
-        expect(PublishingAPI.service(:live_content_store)).to receive(:put_content_item)
-          .with(redirect_response)
+      it "sends a redirect to the live content store" do
+        Timecop.freeze do
+          expect(PublishingAPI.service(:live_content_store)).to receive(:put_content_item)
+            .with(redirect_response)
 
-        post "/v2/content/#{content_id}/unpublish", params: redirect_params
+          post "/v2/content/#{content_id}/unpublish", params: redirect_params
 
-        expect(response.status).to eq(200), response.body
+          expect(response.status).to eq(200), response.body
+        end
       end
-    end
 
-    it "sends a redirect to the draft content store" do
-      Timecop.freeze do
-        expect(PublishingAPI.service(:draft_content_store)).to receive(:put_content_item)
-          .with(redirect_response)
+      it "sends a redirect to the draft content store" do
+        Timecop.freeze do
+          expect(PublishingAPI.service(:draft_content_store)).to receive(:put_content_item)
+            .with(redirect_response)
+
+          post "/v2/content/#{content_id}/unpublish", params: redirect_params
+
+          expect(response.status).to eq(200), response.body
+        end
+      end
+
+      it "does not send to the message queue" do
+        allow(PublishingAPI.service(:live_content_store)).to receive(:put_content_item)
+        allow(PublishingAPI.service(:draft_content_store)).to receive(:put_content_item)
+        expect(PublishingAPI.service(:queue_publisher)).not_to receive(:send_message)
 
         post "/v2/content/#{content_id}/unpublish", params: redirect_params
 
         expect(response.status).to eq(200), response.body
       end
     end
 
-    it "does not send to the message queue" do
-      allow(PublishingAPI.service(:live_content_store)).to receive(:put_content_item)
-      allow(PublishingAPI.service(:draft_content_store)).to receive(:put_content_item)
-      expect(PublishingAPI.service(:queue_publisher)).not_to receive(:send_message)
-
-      post "/v2/content/#{content_id}/unpublish", params: redirect_params
+    context "with a new-style payload" do
```

Comment:
probably want to use a different phrasing here than new and old. Explicitly whats different would be better